tests/components/home_plus_control/test_switch.py

```diff
@@ -0,0 +1,463 @@
+"""Test the Legrand Home+ Control switch platform."""
+import datetime as dt
+from unittest.mock import patch
+
+from homepluscontrol.homeplusapi import HomePlusControlApiError
+
+from homeassistant import config_entries, setup
+from homeassistant.components.home_plus_control.const import (
+    CONF_SUBSCRIPTION_KEY,
+    DOMAIN,
+    ENTITY_UIDS,
+)
+from homeassistant.const import (
+    CONF_CLIENT_ID,
+    CONF_CLIENT_SECRET,
+    STATE_OFF,
+    STATE_ON,
+    STATE_UNAVAILABLE,
+)
+
+from tests.common import async_fire_time_changed
+from tests.components.home_plus_control.conftest import (
+    CLIENT_ID,
+    CLIENT_SECRET,
+    SUBSCRIPTION_KEY,
+)
+
+
+def entity_assertions(
+    hass,
+    num_exp_entities,
+    num_exp_devices=None,
+    expected_entities=None,
+    expected_devices=None,
+):
+    """Assert number of entities and devices."""
+    entity_reg = hass.helpers.entity_registry.async_get(hass)
+    device_reg = hass.helpers.device_registry.async_get(hass)
+
+    if num_exp_devices is None:
+        num_exp_devices = num_exp_entities
+
+    assert len(entity_reg.entities) == num_exp_entities
+    assert len(device_reg.devices) == num_exp_devices
+
+    if expected_entities is not None:
+        for exp_entity_id, present in expected_entities.items():
+            assert bool(entity_reg.async_get(exp_entity_id)) == present
+
+    if expected_devices is not None:
+        for exp_device_id, present in expected_devices.items():
+            assert bool(device_reg.async_get(exp_device_id)) == present
+
+
+def one_entity_assertion(hass, device_uid, availability):
+    """Assert the presence of an entity and its specified availability."""
+    entity_reg = hass.helpers.entity_registry.async_get(hass)
+    device_reg = hass.helpers.device_registry.async_get(hass)
+
+    device_id = device_reg.async_get_device({(DOMAIN, device_uid)}).id
+    entity_entries = hass.helpers.entity_registry.async_entries_for_device(
+        entity_reg, device_id
+    )
+
+    assert len(entity_entries) == 1
+    entity_entry = entity_entries[0]
+    entity_state = hass.states.get(entity_entry.entity_id).state
+    if availability:
```

Comment:
It's not good to have logic in tests. Having this assertion here also makes it harder to read the test and understand what we're testing. If we want to keep this helper have it return the state instead so we can assert correctly in the test.